compiler/luci/pass/src/SubstitutePadV2ToPadPass.cpp

```diff
@@ -83,6 +83,94 @@
 namespace
 {
 
+struct Paddings
+{
+  struct Pad
+  {
+    int32_t front;
+    int32_t end;
+  };
+  /**
+   * @brief Store paddings position information.
+   * @details _padding_pos[k] stores Pad object at axis k
+   */
+  std::vector<Pad> _padding_pos;
+
+  Paddings(luci::CircleConst *paddings)
+  {
+    assert(paddings->dtype() == loco::DataType::S32);
+    assert(paddings->rank() == 2);
+    assert(paddings->dim(1).value() == 2);
+    assert(paddings->size<loco::DataType::S32>() == paddings->rank() * 4);
+
+    for (uint32_t i = 0; i < paddings->dim(0).value(); i++)
+    {
+      Pad pad{.front = paddings->at<loco::DataType::S32>(i * 2),
+              .end = paddings->at<loco::DataType::S32>(i * 2 + 1)};
+      _padding_pos.emplace_back(pad);
+    }
+  }
+
+  /**
+   * @brief Check if this padding area is covered by filter
+   *
+   * @note This is to check condition C4).
+   *       _padding_pos should store values according to NHWC.
+   */
+  bool smaller_than(int32_t filter_h, int32_t filter_w)
+  {
+    assert(_padding_pos.size() == 4);
+
+    auto &pad_H = _padding_pos.at(1);
+    auto &pad_W = _padding_pos.at(2);
+
+    return (pad_H.front < filter_h) && (pad_H.end < filter_h) && (pad_W.front < filter_w) &&
+           (pad_W.end < filter_w);
+  }
+
+  /**
+   * @brief Track how paddings change after CircleTranspose
+   * @details Consider the following graph,
+   *
+   *   %1 = Circle.Input
+   *   %2 = Circle.PadV2(%1,
+   *                     paddings=[[0, 0], [0, 0], [2, 3], [4, 5]],
+   *                     padding_value = -100)
+   *   %3 = Circle.Transpose(%2, perm[0, 2, 3, 1])
+   *
+   *   Output of %3 has padding constant value(-100) from %2 at position below:
+   *
+   *    - axis | front | end
+   *     ------|-------|-----
+   *       0   |   0   |   0
+   *       1   |   2   |   3
+   *       2   |   4   |   5
+   *       3   |   0   |   0
+   *
+   *   This method keeps track of such change of padding position.
+   */
+  void apply(luci::CircleTranspose *transpose)
+  {
+    assert(transpose);
+    luci::CircleConst *perm = loco::must_cast<luci::CircleConst *>(transpose->perm());
+
+    assert(perm && perm->dtype() == loco::DataType::S32);
+    assert(_padding_pos.size() == 4);
+    assert(perm->size<loco::DataType::S32>() == 4);
```

Comment:
Here too.. if these conditions can be false for normal models, it would be better to just return false.
If these MUST be satisfied, it is OK.